parser/prism/Translator.cc

```diff
@@ -513,17 +516,23 @@ ast::ExpressionPtr Translator::desugarMlhs(core::LocOffsets loc, parser::Mlhs *l
     bool didSplat = false;
     auto zloc = loc.copyWithZeroLength();
 
-    for (auto &c : lhs->exprs) {
-        if (auto *splat = parser::NodeWithExpr::cast_node<parser::SplatLhs>(c.get())) {
+    auto lefts = absl::MakeSpan(lhs->lefts.nodes, lhs->lefts.size);
+    auto rights = absl::MakeSpan(lhs->rights.nodes, lhs->rights.size);
+    bool hasSplat = lhs->rest && PM_NODE_TYPE_P(lhs->rest, PM_SPLAT_NODE);
+    size_t totalSize = lefts.size() + (hasSplat ? 1 : 0) + rights.size();
+
+    auto processTarget = [&](pm_node_t *c) {
```

Comment:
I think Stripe had a preference for explicit capture lists. If we do that, it also gives us a chance to capture the smaller read-only values by-value instead of by-ref


```suggestion
    auto processTarget = [this, &didSplat, &stats, &i, &after, &before, totalSize, loc, zloc,
                          tempExpanded](pm_node_t *c) {
```